mongo/integration/sdam_error_handling_test.go

```diff
@@ -85,7 +85,23 @@ func TestSDAMErrorHandling(t *testing.T) {
 				assert.NotNil(mt, err, "expected InsertOne error, got nil")
 				assert.True(mt, mongo.IsTimeout(err), "expected timeout error, got %v", err)
 				assert.True(mt, mongo.IsNetworkError(err), "expected network error, got %v", err)
-				assert.True(mt, tpm.IsPoolCleared(), "expected pool to be cleared but was not")
+				// Assert that the pool is cleared within 2 seconds.
+				assert.Soon(mt, func(ctx context.Context) {
+					ticker := time.NewTicker(100 * time.Millisecond)
```

Comment:
That's correct, the `select` will run whichever `case` unblocks first. While the context isn't "done", it will be the `Ticker` every 100ms. After 2 seconds, the context will be "done" and that `case` will run.